src/main/java/com/epam/grid/engine/service/JobOperationProviderService.java

```diff
@@ -89,6 +94,15 @@ public DeletedJobInfo deleteJob(final DeleteJobFilter deleteJobFilter) {
      * @return Running job.
      */
     public Job runJob(final JobOptions options) {
+        final String workingDir = options.getWorkingDir();
```

Comment:
Could be rewritten to shorten version like:
```
Optional.ofNullable(options.getWorkingDir()).ifPresent(workingDir -> {
            final String properDir = DirectoryPathUtils
                    .buildProperDir(gridSharedFolder, options.getWorkingDir())
                    .toString();
            if (!workingDir.equals(properDir)) {
                options.setWorkingDir(properDir);
                log.info("Working directory was changed from " + workingDir + " to " + properDir);
            }
        });
```

Reply:
done